src/index.js

```diff
@@ -71,84 +61,72 @@ export default function ({Plugin, types: t}) {
         return propPaths.reduce((hash, [keyPath, valuePath]) => {
             let key = getMessageDescriptorKey(keyPath);
 
-            if (DESCRIPTOR_PROPS.has(key)) {
-                let value = getMessageDescriptorValue(valuePath).trim();
-
-                if (key === 'defaultMessage') {
-                    try {
-                        hash[key] = printICUMessage(value);
-                    } catch (e) {
-                        if (isJSXSource &&
-                            valuePath.isLiteral() &&
-                            value.indexOf('\\\\') >= 0) {
-
-                            throw valuePath.errorWithNode(
-                                '[React Intl] Message failed to parse. ' +
-                                'It looks like `\\`s were used for escaping, ' +
-                                'this won\'t work with JSX string literals. ' +
-                                'Wrap with `{}`. ' +
-                                'See: http://facebook.github.io/react/docs/jsx-gotchas.html'
-                            );
-                        }
-
-                        throw valuePath.errorWithNode(
-                            `[React Intl] Message failed to parse: ${e} ` +
-                            'See: http://formatjs.io/guides/message-syntax/'
+            if (!DESCRIPTOR_PROPS.has(key)) {
+                return hash;
+            }
+
+            let value = getMessageDescriptorValue(valuePath).trim();
+
+            if (key === 'defaultMessage') {
+                try {
+                    hash[key] = printICUMessage(value);
+                } catch (parseError) {
+                    if (isJSXSource &&
+                        valuePath.isLiteral() &&
+                        value.indexOf('\\\\') >= 0) {
+
+                        throw valuePath.buildCodeFrameError(
+                            '[React Intl] Message failed to parse. ' +
+                            'It looks like `\\`s were used for escaping, ' +
+                            'this won\'t work with JSX string literals. ' +
+                            'Wrap with `{}`. ' +
+                            'See: http://facebook.github.io/react/docs/jsx-gotchas.html'
```

Comment:
This is actually in the current version, but I changed the function to remove a level of indentation. I think I spent a few hours on this issue only to realize that JSX parses strings like XML/HTML and not JavaScript.